master/buildbot/status/buildstep.py

```diff
@@ -397,7 +405,7 @@ def asDict(self, request=None):
                 for l in self.getLogs()]
 
         result["urls"] = self.getURLs()
-
+        result["urls"].update(self.getArtifactURLs())
```

Comment:
That, and without the UI-end code to support it in place, it wouldn't be much use to include result["artifacts"]

But yes, it is worth keeping in mind that this line will need to be changed later on.